rmgpy/data/thermo.py

```diff
@@ -1251,6 +1261,170 @@ def getThermoData(self, species, trainingSet=None):
+        # only want/need to do one resonance structure,
+        # because will need to regenerate others in gas phase
+        dummyMolecule = molecule.copy(deep=True)
+        sitesToRemove = []
+        for atom in dummyMolecule.atoms:
+            if atom.isSurfaceSite():
+                sitesToRemove.append(atom)
+        for site in sitesToRemove:
+            numbonds = len(site.bonds)
+            if numbonds == 0:
+                #vanDerWaals
+                pass
+            else:
+                assert len(site.bonds) == 1, "Each surface site can only be bonded to 1 atom"
+                bondedAtom = site.bonds.keys()[0]
+                bond = site.bonds[bondedAtom]
+                dummyMolecule.removeBond(bond)
+                if bond.isSingle():
+                    bondedAtom.incrementRadical()
+                elif bond.isDouble():
+                    bondedAtom.incrementRadical()
+                    bondedAtom.incrementRadical()
+                elif bond.isTriple():
+                    bondedAtom.incrementRadical()
+                    bondedAtom.incrementLonePairs()
+                elif bond.isQuadruple():
+                    bondedAtom.incrementLonePairs()
+                    bondedAtom.incrementLonePairs()
+                else:
+                    raise NotImplementedError("Can't remove surface bond of type {}".format(bond.order))
+
+            dummyMolecule.removeAtom(site)
+        dummyMolecule.update()
+
+        logging.debug("Before removing from surface:\n" + molecule.toAdjacencyList())
+        logging.debug("After removing from surface:\n" + dummyMolecule.toAdjacencyList())
+
+        dummySpecies = Species()
+        dummySpecies.molecule.append(dummyMolecule)
+        dummySpecies.generate_resonance_structures()
+        thermo = self.getThermoData(dummySpecies)
+
+        thermo.comment = "Gas phase thermo from {0}. Adsorption correction:".format(thermo.comment)
+        logging.debug("Using thermo from gas phase for species {}\n".format(species.label) + repr(thermo))
+
+        if not isinstance(thermo, ThermoData):
+            thermo = thermo.toThermoData()
+            findCp0andCpInf(species, thermo)
+
+        ## Get the adsorption energy
+        # Create the ThermoData object
+        adsorptionThermo = ThermoData(
+            Tdata = ([300,400,500,600,800,1000,1500],"K"),
+            Cpdata = ([0.0,0.0,0.0,0.0,0.0,0.0,0.0],"J/(mol*K)"),
+            H298 = (0.0,"kJ/mol"),
+            S298 = (0.0,"J/(mol*K)"),
+        )
+        try:
+            self.__addGroupThermoData(adsorptionThermo, self.groups['adsorptionPt'], molecule, {})
+        except KeyError:
+            logging.error("Couldn't find in adsorption thermo database:")
+            logging.error(molecule)
+            logging.error(molecule.toAdjacencyList())
+            raise
+
+        # (groupAdditivity=True means it appends the comments)
+        addThermoData(thermo, adsorptionThermo, groupAdditivity=True)
```

Comment:
Was there a reason for adding the adsorption correction separately instead of providing `thermo` directly to `__addGroupThermoData`?